src/sidebar/components/ThreadList.js

```diff
@@ -96,6 +96,8 @@ function ThreadList({ threads }) {
     // list is the most recently created one.
     const newAnnotations = store
       .unsavedAnnotations()
+      // @ts-expect-error - FIXME: The `isHighlight` check here will always fail
+      // because `unsavedAnnotations` returns objects that do not have a `$highlight` property.
```

Comment:
Improvements to `createSelector` types turned up an existing error here where `unsavedAnnotations` returns objects that come from the `annotation` property of `Draft` instances. These are a subset of `Annotation` objects that only have `id` and `$tag` properties, which is not what `isHighlight` expects. Writing this note made me realize I need to refine this comment a bit.

Writing this note has made me realize that I need to refine this comment, its not just the

Reply:
It turns out the entire `filter` call here was redundant as the callback always returned true. I removed it with notes in a commit message explaining why it had no effect and why functionally everything still worked as it should.